FormalConjectures/Wikipedia/Irrational.lean

```diff
@@ -1,73 +1,187 @@
-open Finset
-
-local notation "e" => exp 1
-
--- See also corresponding transcendence conjectures
--- in `FormalConjectures.Wikipedia.SchanuelsConjecture`
-
-/-- $e + \pi$ is irrational -/
-@[category research open, AMS 33]
-theorem irrational_e_plus_pi : Irrational (e + π) := by
-  sorry
-
-/-- $e \pi$ is irrational -/
-@[category research open, AMS 33]
-theorem irrational_e_times_pi : Irrational (e * π) := by
-  sorry
-
-/-- $e ^ e$ is irrational -/
-@[category research open, AMS 33]
-theorem irrational_e_to_e : Irrational (e ^ e) := by
-  sorry
-
-/-- $\pi ^ e$ is irrational -/
-@[category research open, AMS 33]
-theorem irrational_pi_to_e : Irrational (π ^ e) := by
-  sorry
-
-/-- $\pi ^ \pi$ is irrational -/
-@[category research open, AMS 33]
-theorem irrational_pi_to_pi : Irrational (π ^ π) := by
-  sorry
-
-/-- $\ln(\pi)$ is irrational -/
-@[category research open, AMS 33]
-theorem irrational_ln_pi : Irrational (log π) := by
-  sorry
-
-/-- The the difference between harmonic series and natural logarithm, also known
-as Euler-Mascheroni constant $\gamma$ is irrational -/
-@[category research open, AMS 33]
-theorem irrational_euler_mascheroni :
-    Irrational eulerMascheroniConstant := by
-  sorry
-
-/-- The Catalan constant is irrational -/
-@[category research open, AMS 11, AMS 33]
-theorem irrational_catalan_constant :
-    Irrational (∑' n : ℕ, (-1)^n / (2*n + 1)^2) := by
-  sorry
+/-
+Copyright 2025 The Formal Conjectures Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+-/
+
+import FormalConjectures.Util.ProblemImports
+
+/-!
+# Open questions on irrationality of numbers
+
+*References:*
+- [Wikipedia, Irrational number](https://en.wikipedia.org/wiki/Irrational_number#Open_questions)
+- [Wikipedia, Euler-Lehmer constants](https://en.wikipedia.org/wiki/Euler%27s_constant#Euler-Lehmer_constants)
+- [Wikipedia, Transcendental number](https://en.wikipedia.org/wiki/Transcendental_number)
+-/
+
+open Real Finset Filter
+
+local notation "e" => exp 1
+
+/--
+**Generalized Euler-Lehmer sequence**
+* `a` - residue class modulo `q`
+* `n` - upper bound for summation
+
+The sequence calculates the difference between:
+1. The sum of reciprocals of natural numbers ≤ `n` congruent to `a` modulo `q`
+2. The natural logarithm of `n` scaled by `1/q`
+
+`eulerMascheroniSeq` occurs when `q=1` and `a=1`.
+-/
+noncomputable def eulerLehmerSeq (a : ℕ) (q : ℕ+) (n : ℕ) : ℝ :=
+  let S (x : ℕ) := ∑ k in (Finset.Icc 1 x).filter (· ≡ a [MOD (q : ℕ)]), (1 : ℝ) / k

```

Comment:
Since `S` only gets used once in a line that isn't much more complicated, I think you might as well just inline this